### DISCUSSION TITLE
GBA Wireless Emulation finally possible With Retroarch v1.17 and gpSP, libretro update in onion on next version please!!

### DISCUSSION THREAD

**Discussion by erfanCA01** [OnionUI/Onion#1409] [Ideas]
More Info here : https://www.davidgf.net/2024/01/13/gba-wireless-adapter/
and : https://github.com/libretro/RetroArch/blob/master/CHANGES.md

Been waiting forever for actual multiplayer pokemon, so far it seems to emulate the wireless adapter which is still huge, Hopefully devs can integrate this libretro version update soon!

**Comment by wouter-luimes** [comment]
I've also come across David Guillen Fandos's blog post and was hoping this feature would be implemented in version 4.3.0, but sadly it was not included. Just wanted to bring back this discussion and ask if anybody has any idea if and when this could be added to Onion?

**Reply by erfanCA01** [reply]
The devs are probably busy with thier own work rn, as this is a free project , i think best we can do is learn the development and contribute to the project ourselves 😁

**Reply by jboss38** [reply]
You check this, if you want and try it : https://github.com/OnionUI/Onion/pull/1790

**Reply by wouter-luimes** [reply]
Thanks for sharing the script, but as of the latest beta pre-release from last December, RetroArch has been updated to v1.19.1, which includes the most recent version of gpSP that now supports emulating the Wireless Adapter already.

**Reply by jboss38** [reply]
> Thanks for sharing the script, but as of the latest beta pre-release from last December, RetroArch has been updated to v1.19.1, which includes the most recent version of gpSP that now supports emulating the Wireless Adapter already.

yep for same ROM but not with different ROM (trade between red and green for example)
